litr-filters/src/main/java/com/linkedin/android/litr/filter/video/gl/BaseOverlayGlFilter.java

```diff
@@ -109,8 +125,17 @@ void renderOverlayTexture(int textureId) {
     int createOverlayTexture(@NonNull Bitmap overlayBitmap) {
         int overlayTextureID;
 
+        vertexShaderHandle = GlRenderUtils.loadShader(GLES20.GL_VERTEX_SHADER, VERTEX_SHADER);
+        if (vertexShaderHandle == 0) {
+            throw new RuntimeException("failed loading vertex shader");
+        }
+        fragmentShaderHandle = GlRenderUtils.loadShader(GLES20.GL_FRAGMENT_SHADER, FRAGMENT_OVERLAY_SHADER);
+        if (fragmentShaderHandle == 0) {
+            throw new RuntimeException("failed loading fragment shader");
```

Comment:
vertexShaderHandle will not be free if fragmentShaderHandle failed?

Reply:
Good call. I will `release` the filter on failures, which should all all necessary cleanups.